app/src/main/java/com/eveningoutpost/dexdrip/Models/blueReader.java

```diff
@@ -47,20 +69,27 @@ public static boolean isblueReaderPacket(byte[] buffer) {
         }
         if (bufferstring.startsWith("not ready for") ) { //delete the trans_failed, because its normal only if the bluereader could not read the sensor.
             counterHibernated++;
-            Log.e(TAG, "Found blueReader in a ugly State (" + counterHibernated + "/5), send hibernate to reset! If this does not help in the next 5 Minutes, then turn the bluereader manually off and on!");
-            if (counterHibernated > 4) {
-                Log.wtf(TAG, "Ugly state not resolveable. Please restart the BlueReader! Sending shutdown to blueReader!");
+            Log.e(TAG, "Found blueReader in a ugly State (" + counterHibernated + "/3), send hibernate to reset! If this does not help in the next 5 Minutes, then turn the bluereader manually off and on!");
+            if (counterHibernated > (int)2) {
```

Comment:
both counterHIbernated and 2 are int. You can delete the (int) cast which is not needed.